### DISCUSSION TITLE
Everything works... except audio

### DISCUSSION THREAD

**Discussion by Pilot001** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#519] [Q&A] [answered]
Model: Ibiza (2021)
Pump: Square S100101, 6 wire (selected for pump and display: "6 wire 2021")
PCB: Custom, using a D1 mini as described in [LLC - Why bother... It works fine without...](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/312)
Version: 4.1.3 (but same issue with 3.5.1)

Everything works: display, buttons, web interface, MQTT... except for audio. There is no sound at all - neither during boot nor when a button is pressed - even the SPAconfig says it's on.

Any one any idea?

Thank you in advance!

**Comment by visualapproach** [comment]
Hi, there seems to be a different audio circuit in some 2021 pumps. Some ppl reported weak volume. I don't know what the difference is. You could connect original audio directly without the esp, as a boring work around.

**Reply by Pilot001** [reply]
Hmmm... so it seems I would need to disassemble the display to find out? What do you think?
And about connecting "original audio directly" (I would prefer it different, but just in case): How would I do that? Which of the pins is for audio?

**Comment by visualapproach** [comment, marked as answer]
I guess it wants 5V and you giving it 3.3V may have something to do with it. I wish your coffee would compensate that 😊 Thank you

**Reply by Pilot001** [reply]
You are welcome. :-)
OK, that might be true... so I would need a single level converter on that pin. I will do some measurement and tests on it and let you know.

**Reply by visualapproach** [reply]
To use original audio you just connect wire #6 (opposite from 5V) on cio cable to the same wire on dsp. That wire then don't touch the pcb at all.
One channel LLC would be worth a try, yes.

**Comment by Pilot001** [comment]
Just to finally finish this conversation... you were right and a single LLC does the job. But I don't need other sounds, so as mentioned I reconnected that wire back to the original audio and disabled audio in the menu.
Thanks again!

**Reply by visualapproach** [reply]
Happy it worked out for you! Cheers

**Comment by SeeeeebDe** [comment]
Hello !
I just discovered when needed to update my wifi pasword that there's been a lot of updates since 2021 !

Unfortunatly I have the same issue with no sound. If I understand your solution.. it is to directly connect the audio without passing through the ESP.
But in past firmware (2021...) it used to work fine, with start up melody also (not usefull, but just to say that it was ok without hardware modification).

Isn't there any other solution like juste another pinout to try ?

Sébastien

**Reply by visualapproach** [reply]
@SeeeeebDe Without any info on your pump, pcb or what config you have setup there is not much anyone can say. There is a hardware config page you should setup.

**Reply by SeeeeebDe** [reply]
Oh sorry, you’re right… 🫣
First, thanks for your quick answer.

This is a spa model Miami, same square pump, board v1 built in 2021 when I got the spa.

I went through the hardware setup page but actually I don’t know witch PIN number to try else that the one by default, especially cause I see in the discussion above the this could be an issue about voltage and not only the pinout. But this is disturbing considering that in the old built it was okay.

Another point I’ve noticed is that the target changer is very much slower than before (noticeable especially when changing by several degrees).

**Reply by visualapproach** [reply]
The voltage issue is only if you didn't build according to my manual. I.e. don't have a level converter. But I guess you have since you bothered to buy the PCB. So in the configuration page you just choose pump and display and PCB v1 and you should be good to go. After saving and restarting ofc. The slower setpoint is to avoid overshoots and some pumps were not happy with the faster setting. There are ways to speed it up if you wanna try

**Reply by SeeeeebDe** [reply]
I assembled my board v1 in 2021 with the instructions available at this moment, I’ve that in the current manual there  are some more tips to improve but I didn’t plan to modify what used to work fine until there 😌 I finally shorted the wire nb 6 as you proposed in the posts above.

Yes I would be very interested in trying to speed up the temperature change. 
I‘ve note dived into the code cause I’m not high skilled but I suppose that this is a timer to be set ? If so I should be able to to it by myself if I know were to look at.
How hard would it be to include an additional setting in spa settings page ? 

Thanks again

**Reply by visualapproach** [reply]
In CIO_6W.cpp lower the item.duration_ms until it breaks. Then increase a little. First value is how long the UP/DOWN button is pressed, the next is how long to wait until allowing another press.
```    
    if((cio_toggles.target != cio_states.target) && (_button_que_len == 0))
    {
        unlock();
        Buttons dir;
        cio_toggles.target > cio_states.target ? dir = UP : dir = DOWN;
        item.btncode = getButtonCode(dir);
        item.p_state = &sStates::target;
        item.value = cio_toggles.target;
        item.duration_ms = 800;
        _qButton(item);
        item.btncode = getButtonCode(NOBTN);
        item.p_state = &sStates::char1;
        item.value = 0xFF;
        item.duration_ms = 400;
        _qButton(item);
    }
```

**Reply by SeeeeebDe** [reply]
Hi!

Thank you, again :) And my best wishes for this new year beggining.
I just made quick iterative testings through OTA update.
In my case, the final values I've selected are : 200ms press and 100ms wait durations.

Thank for all the support you offer, just sent you a small contribution :)

______________________________________________

Short observations :
  --> shorter durations brings to some button press missed ; the target not beeing reached keep the loop running until it would finally be ok, but behaviours is appears buggy/laggy, and finally the result could be longer than before optimisation.
  --> in my case the only issue that some time could be seen is a display lag when I jump by a lot of degrees (from 25 to 40 for example), we ear the bip but display could have a small lag ang jump 2-3 degrees at a time after. Not anoying at all cause this is not a realistic use, and anyway I won't be looking at the "countdown" :)
 --> it never results in an under/overshoot, due to "cio_toggles.target != cio_states.target" if condition :) As mentionned before, in the worst case is results in one more loop :)

**Reply by visualapproach** [reply]
Happy new year mate and thank you! Glad I could help.

**Reply by visualapproach** [reply]
> How hard would it be to include an additional setting in spa settings page ?

Depends on what setting you want. What's your thinking?

**Reply by SeeeeebDe** [reply]
Today this might be no use for me since it’s now done, but maybe for the community it could be:
-input for these two variables
-or more guided setting with two options (fast or secure), but it might not be robust if it depends from the material used (pump, pcb,…)

Else, I also thought about an auto-setting mode (that would perform a loop of setting-test to optimize) that could be launched by web ui or buttons combination. But this is definitely another job and effort.
